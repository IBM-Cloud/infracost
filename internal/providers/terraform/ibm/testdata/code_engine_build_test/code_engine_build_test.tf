terraform {
  required_providers {
    ibm = {
      source = "IBM-Cloud/ibm"
    }
  }
}

provider "ibm" {
  region = "us-south"
}

resource "ibm_resource_group" "test_group" {
  name = "test-resource-group"
}

resource "ibm_code_engine_project" "ce_project" {
  name              = "ce_project"
  resource_group_id = ibm_resource_group.test_group.id
}

resource "ibm_code_engine_build" "ce_build" {
  project_id    = ibm_code_engine_project.ce_project.id
  name          = "ce-build"
  output_image  = "private.de.icr.io/icr_namespace/image-name"
  output_secret = "ce-auto-icr-private-eu-de"
  source_url    = "https://github.com/IBM/CodeEngine"
  strategy_type = "dockerfile"
  strategy_size = "small"
}

resource "ibm_code_engine_build" "ce_build2" {
  project_id    = ibm_code_engine_project.ce_project.id
  name          = "ce-build2"
  output_image  = "private.de.icr.io/icr_namespace/image-name"
  output_secret = "ce-auto-icr-private-eu-de"
  source_url    = "https://github.com/IBM/CodeEngine"
  strategy_type = "dockerfile"
}

resource "ibm_code_engine_build" "ce_build3" {
  project_id    = ibm_code_engine_project.ce_project.id
  name          = "ce-build3"
  output_image  = "private.de.icr.io/icr_namespace/image-name"
  output_secret = "ce-auto-icr-private-eu-de"
  source_url    = "https://github.com/IBM/CodeEngine"
  strategy_type = "dockerfile"
  strategy_size = "large"
}

resource "ibm_code_engine_build" "ce_build4" {
  project_id    = ibm_code_engine_project.ce_project.id
  name          = "ce-build4"
  output_image  = "private.de.icr.io/icr_namespace/image-name"
  output_secret = "ce-auto-icr-private-eu-de"
  source_url    = "https://github.com/IBM/CodeEngine"
  strategy_type = "dockerfile"
  strategy_size = "xlarge"
}

resource "ibm_code_engine_build" "ce_build5" {
  project_id    = ibm_code_engine_project.ce_project.id
  name          = "ce-build5"
  output_image  = "private.de.icr.io/icr_namespace/image-name"
  output_secret = "ce-auto-icr-private-eu-de"
  source_url    = "https://github.com/IBM/CodeEngine"
  strategy_type = "dockerfile"
  strategy_size = "xxlarge"
}