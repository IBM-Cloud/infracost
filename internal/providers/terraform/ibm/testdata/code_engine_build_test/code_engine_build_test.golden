
 Name                                        Monthly Qty  Unit        Monthly Cost 
                                                                                   
 ibm_code_engine_build.ce_build                                                    
 ├─ Virtual Processor Cores (small build)            0.5  vCPU Hours         $0.06 
 └─ RAM (small build)                                  2  GB Hours           $0.03 
                                                                                   
 ibm_code_engine_build.ce_build2                                                   
 ├─ Virtual Processor Cores (medium build)             1  vCPU Hours         $0.12 
 └─ RAM (medium build)                                 4  GB Hours           $0.05 
                                                                                   
 ibm_code_engine_build.ce_build3                                                   
 ├─ Virtual Processor Cores (large build)              2  vCPU Hours         $0.25 
 └─ RAM (large build)                                  8  GB Hours           $0.10 
                                                                                   
 ibm_code_engine_build.ce_build4                                                   
 ├─ Virtual Processor Cores (xlarge build)             4  vCPU Hours         $0.49 
 └─ RAM (xlarge build)                                16  GB Hours           $0.20 
                                                                                   
 ibm_code_engine_build.ce_build5                                                   
 ├─ Virtual Processor Cores (xxlarge build)           12  vCPU Hours         $1.48 
 └─ RAM (xxlarge build)                               48  GB Hours           $0.61 
                                                                                   
 OVERALL TOTAL                                                               $3.41 
──────────────────────────────────
7 cloud resources were detected:
∙ 5 were estimated, all of which include usage-based costs, see https://infracost.io/usage-file
∙ 2 were free:
  ∙ 1 x ibm_code_engine_project
  ∙ 1 x ibm_resource_group